pkg/asset/ignition/bootstrap/common.go

```diff
@@ -192,24 +194,40 @@ func (a *Common) generateConfig(dependencies asset.Parents, templateData *bootst
 		}
 	}
 
-	// Check for optional platform specific files/units
+	// Copy platform specific files/units when present with the exception of GCP
+	// On GCP, copy over platform specific files/units only when UserProvisionedDNS is enabled.
```

Comment:
Rather than only copying files when userProvisionedDNS is enabled, it would be better to only conditionally generate the files (and always copy all files). I created an example of how to do this here: https://github.com/sadasu/installer/pull/1

The way this is currently written skips including GCP's custom [report-progress.sh script](https://github.com/openshift/installer/blob/master/data/data/bootstrap/gcp/files/usr/local/bin/report-progress.sh) and I don't think that was the intent.

Reply:
Yup. I had considered templating it but didn't take that route because I hadn't seen [report-progress.sh script](https://github.com/openshift/installer/blob/master/data/data/bootstrap/gcp/files/usr/local/bin/report-progress.sh)

Reply:
Updated.  Slightly different from the suggestion to avoid check for empty templates.

Reply:
> avoid check for empty templates.

Sounds good!